### ISSUE TITLE
["Request"] `Iterable::fold` and `Iterable::combineAll` are duplcates

### ISSUE THREAD

**Issue by brezinajn** [arrow-kt/arrow#2682]
What version are you currently using? 1.0.1

What would you like to see?

`Iterable::fold`
https://github.com/arrow-kt/arrow/blob/main/arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt#L983

and
`Iterable::combineAll`
https://github.com/arrow-kt/arrow/blob/main/arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt#L707

are doing the same and one of them probably should be deprecated.

Original debate on arrow slack: https://kotlinlang.slack.com/archives/C5UPMM0A0/p1646820327599249

**Comment by nomisRev** [comment]
I agree with @brezinajn, I prefer `fold`. `fold` is also the name used in the Kotlin Std so everyone in Kotlin should already be familiar with it.

**Comment by i-walker** [comment]
Will do :D
